config/routes.rb

```diff
@@ -0,0 +1,16 @@
+Rails.application.routes.draw do
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  namespace :api, defaults: { format: :json } do
+    namespace :v1 do
+      resources :items, only: [:index, :show, :create, :destroy]
+      resources :users, only: [:create]
+
+      get '/:username/reservations', to: 'reservations#index'
+      post '/login', to: 'users#login'
```

Comment:
I would keep all my routes RESTful and also my controller action names should be RESTful (having only Index, Show, Create, Update, Destroy as names). Login isn't a RESTful controller action name.

The more RESTful way of doing this is:

```
resources :session, only: [:create]
```

Then i would have a controller named "session_controller" where I'll have a create action.

So the endpoint/url that will be consumed by your front end will be:

POST request to "/sessions"